### ISSUE TITLE
Extremely slow with 100% CPU usage

### ISSUE THREAD

**Issue by HommeSauvage** [vercel/turbo#8801]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

Not easy to reproduce as it's a monorepo with lots of files to build

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

2.0.9

### Describe the Bug

When build a monorepo with 43 packages, turbo is being EXTREMELY slow and basically blocks the computer at 100% usage.

If I run the same build of the whole monorepo with `pnpm -r build`, there is some usage, but the expected kind of CPU usage, it build without issues and I can continue to work.

With turbo, same number of packages to build, it blocks completely the computer with 100% usage. I tried to build with the profiler on, here's the link: https://github.com/DriveFlux/turbo-debug/blob/main/profile.json

### Expected Behavior

To build with roughly the same CPU usage as `pnpm -r build`

### To Reproduce

To have a mixture of a couple of packages with prisma (each package generates its own prisma file) and we use SWC to build.

I would like to share the `profile` first, if that doesn't help, I will try to invest some time in making a complete reproduction.

### Additional context

One funny thing happening with `turbo` versus `pnpm` is the appearance of the little window for a split second for each package being built (It's verifying something it seems, the same kind of window you get when running a new software). This doesn't happen with `pnpm -r build`.

Here's the turbo.json file:
```json
{
	"$schema": "https://turbo.build/schema.json",
	"ui": "tui",
	"globalEnv": ["NPM_TOKEN", "USE_ENV"],
	"tasks": {
		"build": {
			"dependsOn": ["^build", "lingui:compile", "validate-config"],
			"outputs": [".next/**", "!.next/cache/**", "dist/**"]
		},
		"type-check": {
			"cache": false
		},
		"check": {
			"cache": false
		},
		"lint": {
			"cache": false
		},
		"lint:errors": {
			"cache": false
		},
		"format": {
			"cache": false
		},
		"i18n": {
			"inputs": ["**/*.tsx", "**/*.ts"],
			"outputs": ["locales/**/*.mjs"]
		},

		"@org/scheduler#build": {
			"dependsOn": ["^@org/email#build", "^build"],
			"outputs": ["dist/**"]
		},

		"@org/subscription#build": {
			"dependsOn": ["^@org/billing#build", "^build"],
			"outputs": ["dist/**"]
		},

		"dev": {
			"persistent": true,
			"cache": false
		},
		"definitions": {
			"dependsOn": ["schema"],
			"outputs": [".definitions.json"]
		},
		"schema": {},
		"//#watch:packages": {
			"dependsOn": ["//#is-not-watching"],
			"persistent": true,
			"cache": false
		},
		"//#is-not-watching": {
			"cache": false
		},
		"lingui:compile": {
			"inputs": ["locales/**.po"],
			"outputs": ["locales/**.mjs"]
		},
		"validate-config": {
			"cache": false
		}
	}
}
```

**Comment by pawelblaszczyk5** [comment]
Out of curiosity, does it happen with the old ui as well? Because I have similar issue that was fixed by disabling the new one.

**Comment by chris-olszewski** [comment]
> One funny thing happening with turbo versus pnpm is the appearance of the little window for a split second for each package being built

Can you elaborate on this? This seems highly suspect. Are you referring to a macOS popup window like this: 
![image](https://github.com/user-attachments/assets/6f23a32f-20cd-4cb0-a5a3-edf078d3d491)

**Comment by HommeSauvage** [comment]
No, it's difficult to catch the little window, it's just verifying a bin. I had to do a screen recording to catch the window as it disappears immediately.

The bin is a node module from the monorepo. The funny thing is that macOS was not verifying it and just runs it. However, when we run it with Turbo UI, it does the verification. It could be due to the updates of swc? But again, only turbo UI seems to trigger this. Also, the XProtectService process works a lot.

Subsequent builds (even when there's no cache) do not have this problem.

<img width="522" alt="Screenshot 2024-07-26 at 1 20 58 PM" src="https://github.com/user-attachments/assets/ccca27f2-1020-464d-9409-dba5a79c97ec">

**Comment by HommeSauvage** [comment]
Removing shebang didn't change much. Removing the `ui` definitely makes the build faster and uses less resources. Notice the difference of usage (the spike in between is just the start of my build process and checks). See before (ui) and after (no ui) the spike.

With ui, `XProtectService` process uses a lot more CPU.

<img width="213" alt="Screenshot 2024-07-26 at 1 36 29 PM" src="https://github.com/user-attachments/assets/affab8fb-63a1-4db7-9179-959d411ba24d">

**Comment by anthonyshew** [comment]
Going to open this back up because we had to revert in #12507. The real fix we should implement here is getting our macOS binaries signed, so I'll work on that.

**Comment by anthonyshew** [comment]
Linking #6935 so I don't forget to close this, too, once its been handled.

**Comment by gregberge** [comment]
Hey @anthonyshew it seems fixed in the latest release for me.

**Comment by anthonyshew** [comment]
What you were experiencing was most likely this: https://github.com/vercel/turborepo/issues/12693

This is still a relevant issue, from what I can tell. I still haven't gotten these binaries signed today.

**Comment by anthonyshew** [comment]
Alright, folks, I *think* this will get solved with [2.9.15-canary.3](https://www.npmjs.com/package/turbo/v/2.9.15-canary.3), that being the first version of Turborepo with signed and notarized binaries for macOS.

Please let me know if that doesn't resolve this and I can circle back to look for a new fix attempt.
